examples/experimental/dagster-airlift/dagster_airlift/core/sensor/builder.py

```diff
@@ -132,13 +144,36 @@ def airflow_dag_sensor(context: SensorEvaluationContext) -> SensorResult:
                 end_date_lte=None,
                 dag_query_offset=0,
             )
+        if event_transformer_fn:
+            with user_code_error_boundary(
+                AirliftSensorEventTransformerError,
+                lambda: f"Error occurred during event transformation for {airflow_data.airflow_instance.name}",
+            ):
+                updated_asset_events = list(
+                    event_transformer_fn(context, airflow_data, all_asset_events)
+                )
+
+            for asset_event in updated_asset_events:
+                if not isinstance(
+                    asset_event, (AssetMaterialization, AssetObservation, AssetCheckEvaluation)
+                ):
+                    raise DagsterInvariantViolationError(
+                        f"Event transformer function must return AssetMaterialization, AssetObservation, or AssetCheckEvaluation objects. Got {type(asset_event)}."
+                    )
+                if EFFECTIVE_TIMESTAMP_METADATA_KEY not in asset_event.metadata:
+                    raise DagsterInvariantViolationError(
+                        f"All returned events must have an effective timestamp, but {asset_event} does not. An effective timestamp can be used by setting dagster_airlift.constants.EFFECTIVE_TIMESTAMP_METADATA_KEY with a dagster.TimestampMetadataValue."
+                    )
```

Comment:
probably worth extracting all this slop into a sep function to clean up the core loop

Reply:
slop: extracted